packages/react-environment/src/WebbProvider.tsx

```diff
@@ -106,20 +113,24 @@ export const WebbProvider: FC<WebbProviderProps> = ({ applicationName = 'Webb Da
         });
         const web3Provider = await Web3Provider.fromWalletConnectProvider(provider);
         await web3Provider.eth.net.isListening();
-        const net = await web3Provider.netowrk;
-        const chainType = WebbWeb3Provider.chainType(net); //  use this to pick the storage
-        const rainkybeS = await Storage.newFresh(WebbEVMChain.Rinkeby, rankebyStorage);
-        const mainS = await Storage.newFresh(WebbEVMChain.Main, mainStorage);
+        const net = await web3Provider.network; // storage based on network id
+        const chainType = WebbWeb3Provider.storageName(net);
         let storage: Storage<EVMStorage>;
-        switch (chainType) {
-          case WebbEVMChain.Main:
-            storage = mainS;
-            break;
-          case WebbEVMChain.Rinkeby:
-            storage = rainkybeS;
-            break;
-        }
-        const webbWeb3Provider = await WebbWeb3Provider.init(web3Provider, rainkybeS);
+          switch (net) {
```

Comment:
It's good to init storages lazily, hover other storages are needed for the provider if the users decided to change the network.